mongodbatlas.go

```diff
@@ -122,6 +137,7 @@ func (m *MongoDBAtlas) NewUser(ctx context.Context, req dbplugin.NewUserRequest)
 		DatabaseName: databaseUser.DatabaseName,
 		Roles:        databaseUser.Roles,
 		Scopes:       databaseUser.Scopes,
+		X509Type:     databaseUser.X509Type,
```

Comment:
Would you mind explaining what the purpose of this field is?  I see from the tests we can specify a type like `CUSTOMER`, but how does that influence the type of certificate being generated? I tried looking at the Vault PR as well but wasn't able to discern 🤔

Reply:
Thanks for explaining!